radiation/python/radsw/radsw_main.py

```diff
@@ -0,0 +1,4031 @@
+                )
+            elif self.isubcsw == 2:
+                print(
+                    "   --- Using MCICA sub-colum clouds approximation ",
+                    "   with provided input array of permutation seeds",
+                )
+            else:
+                print(
+                    "  *** Error in specification of sub-column cloud ",
+                    f" control flag isubcsw = {self.isubcsw} !!",
+                )
+
+        #  --- ...  check cloud flags for consistency
+
+        if (icldflg == 0 and iswcliq != 0) or (icldflg == 1 and iswcliq == 0):
+            print(
+                "*** Model cloud scheme inconsistent with SW",
+                " radiation cloud radiative property setup !!",
+            )
+
+        if self.isubcsw == 0 and self.iovrsw > 2:
+            if me == 0:
+                print(
+                    f"*** IOVRSW={self.iovrsw} is not available for",
+                    " ISUBCSW=0 setting!!",
+                )
+                print("The program will use maximum/random overlap", " instead.")
+            self.iovrsw = 1
+
+        #  --- ...  setup constant factors for heating rate
+        #           the 1.0e-2 is to convert pressure from mb to N/m**2
+
+        if iswrate == 1:
+            self.heatfac = con_g * 864.0 / con_cp  #   (in k/day)
+        else:
+            self.heatfac = con_g * 1.0e-2 / con_cp  #   (in k/second)
+
+        #  --- ...  define exponential lookup tables for transmittance. tau is
+        #           computed as a function of the tau transition function, and
+        #           transmittance is calculated as a function of tau.  all tables
+        #           are computed at intervals of 0.0001.  the inverse of the
+        #           constant used in the Pade approximation to the tau transition
+        #           function is set to bpade.
+
+        self.exp_tbl = np.zeros(ntbmx + 1)
+        self.exp_tbl[0] = 1.0
+        self.exp_tbl[ntbmx] = expeps
+
+        for i in range(ntbmx - 1):
+            tfn = i / (ntbmx - i)
+            tau = self.bpade * tfn
+            self.exp_tbl[i] = np.exp(-tau)
+
+    def return_initdata(self):
+        outdict = {"heatfac": self.heatfac, "exp_tbl": self.exp_tbl}
+        return outdict
+
+    def swrad(
+        self,
+        plyr,
+        plvl,
+        tlyr,
+        tlvl,
+        qlyr,
+        olyr,
+        gasvmr,
+        clouds,
+        icseed,
+        aerosols,
+        sfcalb,
+        dzlyr,
+        delpin,
+        de_lgth,
+        cosz,
+        solcon,
+        NDAY,
+        idxday,
+        npts,
+        nlay,
+        nlp1,
+        lprnt,
+        lhswb,
+        lhsw0,
+        lflxprf,
+        lfdncmp,
+    ):
+
+        self.lhswb = lhswb
+        self.lhsw0 = lhsw0
+        self.lflxprf = lflxprf
+        self.lfdncmp = lfdncmp
+
+        # outputs
+        hswc = np.zeros((npts, nlay))
+        cldtau = np.zeros((npts, nlay))
+
+        upfxc_t = np.zeros(npts)
```

Comment:
not required for this PR, but since you have an 'init' and a 'call' method, you could define all these arrays inside of the init so that they don't get created every timestep. But clearly still in python loops, you don't need to worry about that yet.

Reply:
good idea, thanks!